### PR TITLE
Create auth configmap containing MC name and admin group info

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards https://github.com/giantswarm/roadmap/issues/2610

This creates a configmap for the cluster containing MC name and MC giantswarm admin groups. This configmap can be included in extraConfigs for auth bundle (similar to the one for the observability bundle) so that cluster role binding for admin groups are created and athena includes mc name (so that unique context name can be created for login).

### CODE REVIEW

File: pkg/auth/.DS_Store
Comment (kopiczko) [inline pkg/auth/.DS_Store:1]: Oops

File: helm/dex-operator/Chart.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ icon: https://s.giantswarm.io/app-icons/dex/2/icon_light.svg
 apiVersion: v1
 annotations:
   application.giantswarm.io/team: "bigmac"
-  config.giantswarm.io/version: 1.x.x
+  config.giantswarm.io/version: auth-cm
```
Comment (kopiczko) [inline helm/dex-operator/Chart.yaml:10]: Is that intentional?

**Review by kopiczko — APPROVED** [review]
Two small comments, but LGTM in general

File: helm/dex-operator/Chart.yaml (hunk 1 above)
Comment (anvddriesch) [reply helm/dex-operator/Chart.yaml:10]: I used it for testing. :D I'll get rid of it before merge

**Comment by anvddriesch** [comment]
Hmm I see what you mean.
Right now the operator reconciles dex apps and creates configuration for them. (So it does not actually create the dex app but it configures it).
Since it's creating a secret with idp configuration it does not feel too weird for it to also create the configmap with auth configuration.
However, the name really isn't ideal in this case. `auth-operator` would be better. Also, it'd be better for it to reconcile the auth-bundle and not the dex app. But we would need to do that when the auth-bundle is actually present.
Changing the name of an operator we already use on all providers is cumbersome, though. :grimacing:

**Comment by anvddriesch** [comment]
We discussed this today in the refinement and decided to go ahead with it for now.
However, we will rename the dex-operator to auth-operator in the future and it will be reconciling the auth-bundle.

**Pull request merged**